Mathlib/Geometry/Synthetic/Avigad/Axioms.lean

```diff
@@ -0,0 +1,245 @@
+      ¬SameSide a b L → SameSide a c L ∨ SameSide b c L
+    /--Relations between sidedness and betweeness-/
+    sameSide12_of_B123_sameSide13 : ∀ {a b c L}, B a b c → SameSide a c L → SameSide a b L
+    /--Relations between sidedness and betweeness-/
+    sameSide_of_B_not_onLine_2 : ∀ {a b c L}, B a b c → OnLine a L → ¬OnLine b L → SameSide b c L
+    /--Relations between sidedness and betweeness-/
+    not_sameSide13_of_B123_onLine2 : ∀ {a b c L}, B a b c → OnLine b L → ¬SameSide a c L
+    /--Relations between sidedness and betweeness-/
+    B_of_onLine_inter : ∀ {a b c L M}, a ≠ b → b ≠ c → a ≠ c → L ≠ M →
+      OnLine a L → OnLine b L → OnLine c L → OnLine b M → ¬SameSide a c M → B a b c
+    /--Deducing sidedness from three lines intersecting-/
+    not_sameSide_of_sameSide_sameSide : ∀ {a b c d L M N}, OnLine a L → OnLine a M → OnLine a N →
+      OnLine b L → OnLine c M → OnLine d N → SameSide c d L → SameSide b c N → ¬SameSide b d M
+    /--Deducing sidedness from three lines intersecting-/
+    sameSide_of_sameSide_not_sameSide : ∀ {a b c d L M N}, a ≠ b → OnLine a L → OnLine a M →
+      OnLine a N → OnLine b L → OnLine c M → OnLine d N → ¬OnLine d M → SameSide c d L →
+      ¬SameSide b d M → SameSide b c N
+    /--Points of intersection of a line and circle-/
+    B_of_lineCircleInter : ∀ {a b c L α}, b ≠ c → OnLine a L →
+      OnLine b L → OnLine c L → OnCircle b α → OnCircle c α → InCircle a α → B b a c
+    /--The points of intersection of circles are on opposite sides the line that joins the centers-/
+    not_sameSide_of_circle_inter : ∀ {a b c d L α β},  c ≠ d → α ≠ β →  OnLine a L → OnLine b L →
+      OnCircle c α → OnCircle c β → OnCircle d α → OnCircle d β → CenterCircle a α →
+      CenterCircle b β → CirclesInter α β → ¬SameSide c d L
+    /--Condition for lines intersecting-/
+    lines_inter_of_not_sameSide : ∀ {a b L M}, OnLine a M → OnLine b M → ¬SameSide a b L →
+      LinesInter L M
+    /--Condition for line circle intersection-/
+    lineCircleInter_of_not_sameSide : ∀ {a b L α}, ¬SameSide a b L → OnCircle a α ∨ InCircle a α →
+      OnCircle b α ∨ InCircle b α → LineCircleInter L α
+    /--Condition for line circle intersection-/
+    lineCircleInter_of_inside_onLine : ∀ {a L α}, OnLine a L → InCircle a α → LineCircleInter L α
+    /--Condition for circle circle intersection-/
+    circlesInter_of_inside_on_circle : ∀ {a b α β}, OnCircle b α → OnCircle a β → InCircle a α →
+      InCircle b β → CirclesInter α β
+    /--A length is zero iff the points are equal-/
+    length_eq_zero_iff : ∀ {a b}, length a b = 0 ↔ a = b
+    /--Length is symmetric-/
+    length_symm : ∀ a b, length a b = length b a
+    /--Angles are symmetric across the middle point-/
+    angle_symm : ∀ a b c, angle a b c = angle c b a
+    /--Angles are nonnegative-/
+    angle_nonneg : ∀ a b c, 0 ≤ angle a b c
+    /--Lengths are nonnegative-/
+    length_nonneg : ∀ a b, 0 ≤ length a b
```

Comment:
I see that you don't require the triangle inequality here. What implies it? Can we `extend MetricSpace X` instead of introducing domain-specific `length`? Note that the goal of Mathlib is to have different parts of the library play well with each other.